indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/StreamChunkParser.java

```diff
@@ -54,12 +64,16 @@
       @Nullable InputFormat inputFormat,
       InputRowSchema inputRowSchema,
       TransformSpec transformSpec,
-      File indexingTmpDir
+      File indexingTmpDir,
+      Predicate<InputRow> rowFilter,
```

Comment:
I don't think we should use a `Predicate<InputRow>` here and in other similar places. Throwing a `ParseException` is expected behavior, so I think callers should know that they need to handle that exception or throw it themselves.

It's not obvious to me how the caller in `FilteringCloseableInputRowIterator` should know they should handle a parse exception https://github.com/apache/druid/pull/10336/files#diff-b7f61f3d28afdd25bedf2efa607fdf88R67

How can I check that the ParseException is handled at the correct level everywhere? I realize this is challenging because ParseException is a RuntimeException.

Reply:
> I don't think we should use a Predicate<InputRow> here and in other similar places. Throwing a ParseException is expected behavior, so I think callers should know that they need to handle that exception or throw it themselves.
>
> It's not obvious to me how the caller in FilteringCloseableInputRowIterator should know they should handle a parse exception https://github.com/apache/druid/pull/10336/files#diff-b7f61f3d28afdd25bedf2efa607fdf88R67

Hmm, I'm not sure I understand your concern. Can you elaborate more?. `rowFilter` is a simple predicate to filter out unnecessary rows and does nothing with parseExceptions. Instead, it just passes the ParseException to the caller if it is thrown in `test()`. The intention here is that the callers of `FilteringCloseableInputRowIterator` can iterate rows without worrying about validation of rows or parseExceptions.